dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/datafile/page/FileDownloadRequestHelper.java

```diff
@@ -0,0 +1,71 @@
+package edu.harvard.iq.dataverse.datafile.page;
+
+import edu.harvard.iq.dataverse.datafile.FilePermissionsService;
+import edu.harvard.iq.dataverse.license.dto.RestrictedTermsOfUseDTO;
+import edu.harvard.iq.dataverse.persistence.datafile.DataFile;
+import edu.harvard.iq.dataverse.persistence.datafile.license.FileTermsOfUse;
+
+import javax.ejb.EJB;
+import javax.faces.view.ViewScoped;
+import javax.inject.Named;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@ViewScoped
+@Named("FileDownloadRequestHelper")
+public class FileDownloadRequestHelper implements Serializable {
+    
+    @EJB
+    private FilePermissionsService filePermissionsService;
+
+    private List<DataFile> filesForRequestAccess = new ArrayList<DataFile>();
+    
+    
+    public void addFileForRequestAccess(DataFile dataFile) {
+        this.filesForRequestAccess.clear();
+        this.filesForRequestAccess.add(dataFile);
+    }
+
+    public void clearRequestAccessFiles() {
+        this.filesForRequestAccess.clear();
+    }
+
+    public void addMultipleFilesForRequestAccess(DataFile dataFile) {
```

Comment:
Maybe the root of the problem isn't in this method naming alone?
How about if I will rename:
`addFileForRequestAccess` to `replaceRequestAccessWithSingleFile` and
`addMultipleFilesForRequestAccess` to `addFileForRequestAccess`? Then it would be similar to list like operations.

Reply:
ok